### DISCUSSION TITLE
clippig netcdf using rioxarray

### DISCUSSION THREAD

**Discussion by gkb999** [corteva/rioxarray#555] [Q&A] [answered]
Hello team,

The following is my NC input bounds as follows:
image
![image](https://user-images.githubusercontent.com/7091088/182982052-1e4f0fb0-0a16-45ab-acce-6912fb510510.png)

My guess: it includes global values.

What I'm looking at: Clipping for the following bounds (in lat-lon)
min_lon = 140 & min_lat = -90 & max_lon = 220 &max_lat = -65 (Southern hemisphere)

Adding to this, my projections are in linear units: meters,
My input nc's projection is as below:
![image](https://user-images.githubusercontent.com/7091088/182982070-2295c0ae-c972-488a-bf8f-f9fe2221133b.png)

I need to clip for the said lat-lon.
As I don't have idea what these values mean in meters, I used utm.from_latlon(-65, 140): to get a particular value in meters and then tried clipping as da.rio.clip_box(minx= 2035.06, miny=500000.00, maxx=2791172.45, maxy= 547155.12,)

NOW, THE CLIPPED ARRAY HAS: 
![image](https://user-images.githubusercontent.com/7091088/182982121-d98c1fd4-6fc4-447f-95b9-360f1f6ec1dd.png)

Does this mean 3 longitudes and 112 latitudes???

Then, this is undoubtedly the wrong way.
Can someone please suggest me the right way?

What I also tried:

1)Converting XY to lat-lon and then select from data array: DID NOT WORK
2) Transforming the projection of the NC data using TRANSFORMER.TRANSFORM: NOT SUCCESSFUL AGAIN
3)Last method, getting lat-lo values into meters, then feed as input for the bounding box - DIDNT WORK AGAIN.
Please provide me with the appropriate way.
I'm sure I have messed up using multiple ways and am half-lost already.

**Comment by snowman2** [comment, marked as answer]
Have you tried this?

https://pyproj4.github.io/pyproj/stable/api/transformer.html#pyproj.transformer.Transformer.transform_bounds

```python
transformer = Transformer.from_crs("EPSG:4326", data.rio.crs, always_xy=True)
data.rio.clip_box(*transformer.transform_bounds(140, -90, 220, -65))
```

**Reply by gkb999** [reply]
It returns the following
![image](https://user-images.githubusercontent.com/7091088/183228236-098e1934-5c75-4d9e-90c2-58ef43d25066.png)

This is the datum.
Should set crs before this?
Something like : data.rio.set_crs("EPSG:4326")? or data.rio.write_crs("epsg:4326", inplace=True)?

**Reply by snowman2** [reply]
The information you provided above is not sufficient to build a CRS. You will need to find more information in order to build the CRS.

**Reply by gkb999** [reply]
This is a raw data projection
![image](https://user-images.githubusercontent.com/7091088/183228605-b9c73fae-5b32-41e2-89a8-f728cb81a3b7.png)

proj_crs_code :
EPSG:3412

![image](https://user-images.githubusercontent.com/7091088/183229177-2a440afa-9606-42f6-86f1-e9f9738792b5.png)

Most of the time 
data.rio.crs return nothing. At once, after a long wait it returned EPSG:3976, but when tried again it returned.

Hence, what I tried now:
![image](https://user-images.githubusercontent.com/7091088/183229300-04577e27-16e5-4ec4-b89b-a9647a23c261.png)
![image](https://user-images.githubusercontent.com/7091088/183229310-7485a3e2-aa9a-43d2-8a9f-d9cc62733abf.png)

I hope this information is better than the previous one

**Reply by gkb999** [reply]
I changed the spatial extent from 220 to -140 because of following error:
![image](https://user-images.githubusercontent.com/7091088/183229421-90656ce3-c847-4634-b5de-f0d4b70fd74a.png)

**Reply by snowman2** [reply]
Your transformer needs to be:
```python
transformer = Transformer.from_crs("EPSG:4326", "EPSG:3412", always_xy=True)
```
This is because your input coordinates are geographic and your output bounds need to be in the CRS of your raster.

**Reply by gkb999** [reply]
This gives me 
![image](https://user-images.githubusercontent.com/7091088/183229762-3b5b871a-ee60-4c67-a038-d458422bf6bd.png)

and i set CRS using: data.rio.write_crs("EPSG:4326")
![image](https://user-images.githubusercontent.com/7091088/183229813-8e8586e7-e55e-4e9e-899b-304355a9ada8.png)
that again produces same error
![image](https://user-images.githubusercontent.com/7091088/183229825-c9a5824e-c817-4f61-bbaa-7541ad8501b9.png)

**Reply by gkb999** [reply]
wow, after adding inplace = true 
![image](https://user-images.githubusercontent.com/7091088/183229894-dc70451f-8f8a-432e-bce2-7b86b2eac2ca.png)

It seems it worked, need to test if it is on the right spot.
Thanks a ton!!!

**Reply by snowman2** [reply]
The raster CRS need to be set to: `EPSG:3412`
```python
data.rio.write_crs("EPSG:3412", inplace=True)
```

**Reply by gkb999** [reply]
Thanks for the correction, 
Yet, that doesn't seem to change any.
Please refer to the following
![image](https://user-images.githubusercontent.com/7091088/183230260-6c42295e-ddfc-45dc-9fb3-3a66159ee8f0.png)
Also, why is spatial_ref null : ()

**Reply by snowman2** [reply]
> why is spatial_ref null : ()

See: https://corteva.github.io/rioxarray/stable/getting_started/crs_management.html

**Reply by gkb999** [reply]
Thanks a lot, that makes sense now
